clash-vexriscv/src/VexRiscv/FFI.hsc

```diff
@@ -32,13 +36,30 @@ foreign import ccall unsafe "vexr_jtag_bridge_init" vexrJtagBridgeInit :: Word16
 foreign import ccall unsafe "vexr_jtag_bridge_step" vexrJtagBridgeStep :: Ptr VexRiscvJtagBridge -> Ptr JTAG_OUTPUT -> Ptr JTAG_INPUT -> IO ()
 foreign import ccall unsafe "vexr_jtag_bridge_shutdown" vexrJtagBridgeShutdown :: Ptr VexRiscvJtagBridge -> IO ()
 
+-- TODO: Move to clash-prelude?
+deriving instance Uniform Bit
+instance UniformRange Bit where
+  uniformRM (Bit lMask lDat, Bit hMask hDat) g = do
+
+    mask <- uniformRM (lMask, hMask) g
+    dat <- uniformRM (lDat, hDat) g
+    pure $ Bit mask dat
+
+instance Random Bit where
+  random g = (Bit 0 a, g')
+   where
+    (a, g') = (randomR (0, 1) g)
```

Comment:
This isn't a random bit, it's a random _defined_ bit. I think I'd just make this a separate function:

```haskell
genDefinedBit :: IO Bit
genDefinedBit = _
```

Also see comment below.